icon-lang: 9.5.1 -> unstable-2020-02-05

KoviRobi opened the pull request:
Latest release (https://github.com/gtownsend/icon/releases/tag/ok2018j)
doesn't seem like a properly versioned release. Latest properly
versioned release
(https://github.com/gtownsend/icon/releases/tag/rel951) is from 2013. So
I have decided to use the unstable release, which works.

<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change

(This was ZHF: #80379 but a different PR resolved that)

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`

   Not sure I have done this correctly, it doesn't seem to build e.g. `noweb`, which I know `noweb` depends on this (and is currently broken). Will try to make `noweb` work too, a `grep` doesn't indicate any other dependencies.

- [ ] Tested execution of all binary files (usually in `./result/bin/`)

    Not tested all binary files, but tested noweb (see PR #81042, which depends on this)

- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Commit:
icon-lang: refactor doc location

yurrriq commented:
LGTM thanks!

xfix commented:
I think patching the build instead of using unstable release may make more sense: https://github.com/NixOS/nixpkgs/pull/80714

KoviRobi commented:
Possibly, but version 9.5.1 is seven years old, which is why I decided to do so. Perhaps if you want to stick with releases, `ok2018j` might be better. But anyway, I didn't notice someone else also fixed this; I fixed noweb with #81042 which should work on top of your now merged fix too (hopefully, I haven't tested it yet).

SuperSandro2000 commented:
Please resolve the merge conflict(s).

SuperSandro2000 commented:
Please resolve the merge conflict(s).

Commit:
icon-lang: 9.5.1 -> unstable-2020-02-05

Latest release (https://github.com/gtownsend/icon/releases/tag/ok2018j)
doesn't seem like a properly versioned release. Latest properly
versioned release
(https://github.com/gtownsend/icon/releases/tag/rel951) is from 2013. So
I have decided to use the unstable release, which works.

SuperSandro2000 commented:
Result of `nixpkgs-review pr 81038` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>icon-lang</li>
    <li>noweb</li>
  </ul>
</details>
